### ISSUE TITLE
Noise being generated right from the start of the training

### ISSUE THREAD

**Issue by janzd** [bghira/SimpleTuner#302]
I encountered a new issue now.
When I start training, the model only generates noise.
After the very first iteration, it generates basically an empty image 
![image](https://github.com/bghira/SimpleTuner/assets/5581222/b4efeeba-7656-4fe5-a357-d2f3876e140a)
5 iterations in and I get noise.
![image](https://github.com/bghira/SimpleTuner/assets/5581222/a27c9023-1d38-4037-935a-69eb1e850ea8)
Do you have any idea what could be going wrong?
I'm not really sure what parameters in the settings could be relevant because it's not that the model training goes wrong because of learning rate for example as the generated results are weird before any training actually happens.

I use this as my validation settings.
```
export VALIDATION_PROMPT="A picture of a tall man."
export VALIDATION_GUIDANCE=7.5
# You'll want to set this to 0.7 if you are training a terminal SNR model.
export VALIDATION_GUIDANCE_RESCALE=0.0
# How frequently we will save and run a pipeline for validations.
export VALIDATION_STEPS=1
export VALIDATION_NUM_INFERENCE_STEPS=30
export VALIDATION_NEGATIVE_PROMPT="blurry, cropped, ugly"
export VALIDATION_SEED=42
export VALIDATION_RESOLUTION=$RESOLUTION
```
This is the base model
```
export MODEL_NAME="stabilityai/stable-diffusion-xl-base-1.0"
```

Debug message says that it does use the caption set in the config file.
```
Hashing caption: A picture of a tall man.
2024-02-07 10:20:34,481 [DEBUG] (TextEmbeddingCache) (Rank: 0) (id=<ID>) compute_embeddings_for_sdxl_prompts received list of prompts: ['A picture of a tall man.']
2024-02-07 10:20:34,481 [DEBUG] (TextEmbeddingCache) Hashing caption: A picture of a tall man.
2024-02-07 10:20:34,482 [DEBUG] (LocalDataBackend) Checking if /data/src/SimpleTuner/data/<DATA DIR>/textembed_cache/a8cdd2c2e30736bfd729b8f14a2937e3-sdxl.pt exists = True
2024-02-07 10:20:34,989 [DEBUG] (validation) Generating validation image: A picture of a tall man.
 -> Number of images: 1
 -> Number of inference steps: 30
 -> Guidance scale: 7.5
 -> Guidance rescale: 0.0
 -> Resolution: 1024.0
 -> Extra validation kwargs: {'generator': <torch._C.Generator object at 0x7f5a7c0cc890>}
2024-02-07 10:20:39,631 [DEBUG] (validation) Completed generating image: A picture of a tall man.
```

**Comment by bghira** [comment]
do you use weights and biases? the whole command run by the script is there on teh overview page. however if not, you can simply make the `train_sdxl.sh` script print out the command it will run instead of running it. but either way, it would be helpful to inspect the rules in play.

i use simpletuner for training v-prediction models with zero-terminal SNR. if you base it on `ptx0/terminus-xl-gamma-v2` instead of SDXL 1.0 it will possibly work? but if that is the case, then there is a bug in how it detects and handles the model config.

**Comment by zhuliyi0** [comment]
I had the same issue, switching to the ztsnr base model fixed it.

**Comment by janzd** [comment]
This is the command for training.
```
accelerate launch  --mixed_precision=bf16 --num_processes=1 --num_machines=1 --dynamo_backend=no train_sdxl.py --pretrained_model_name_or_path=stabilityai/stable-diffusion-xl-base-1.0 --enable_xformers_memory_efficient_attention --gradient_checkpointing --set_grads_to_none --gradient_accumulation_steps=4 --resume_from_checkpoint=  --snr_gamma=5 --data_backend_config=/data/src/SimpleTuner/data/user300k_sr.json --num_train_epochs=25 --max_train_steps=30000 --metadata_update_interval=65 --learning_rate=8e-7 --lr_scheduler=constant --seed 420420420 --lr_warmup_steps=3000 --output_dir=/data/src/SimpleTuner/ckpt/sdxl_finetune_run2 --inference_scheduler_timestep_spacing=trailing --training_scheduler_timestep_spacing=trailing --report_to=tensorboard       --allow_tf32 --mixed_precision=bf16 --vae_dtype=bf16  --train_batch=6 --caption_dropout_probability=0.1 --validation_prompt=A picture of a tall man. --num_validation_images=1 --validation_num_inference_steps=30 --validation_seed=42 --minimum_image_size=1024 --resolution=1024 --validation_resolution=1024 --resolution_type=pixel --checkpointing_steps=250 --checkpoints_total_limit=2 --validation_steps=100 --tracker_run_name=simpletuner-sdxl --tracker_project_name=sdxl-training --validation_guidance=7.5 --validation_guidance_rescale=0.0 --validation_negative_prompt=blurry, cropped, ugly 
```

I tried changing the base model to `tx0/terminus-xl-gamma-v2` but the result is the same.

**Comment by bghira** [comment]
please test with v0.9.0-rc8

**Comment by janzd** [comment]
Sorry, I was away for a few days, so I couldn't test it.
It seems to be okay now.
I'm still not sure if it's converging properly to the data distribution, it's either much slower than it was in a previous version, or there's still some issue, but I'll give it a few more tries and see if it works or not.

**Comment by janzd** [comment]
I'm using 8bit AdamW now with the default learning rate.
When I trained a model with default settings at the beginning of January using the version of the code that was up-to-date back then, the generated results actually matched the what the training data looked like, but now I can't really achieve that with the default settings anymore. The base model and data are the same as then.

This is output after 10 iterations
![image](https://github.com/bghira/SimpleTuner/assets/5581222/5015c0e3-30ed-4144-982f-8218b1184649)
And this is after 4000 iterations
![image](https://github.com/bghira/SimpleTuner/assets/5581222/e55f455b-51c1-42c9-8b83-e278586c4a81)
It doesn't really change.

Btw the training data are chibi characters, so the style of what is being generated is very different from the training data.

I'll play around a bit more to see if I can make it properly learn and match the style/distribution of the training data again.

**Comment by bghira** [comment]
are you using LoRA, or full for `model_type`?

**Comment by janzd** [comment]
I've tried various settings but the results didn't change much.
I use full model finetuning, but I have tried only LoRA too and the results were similar.
Even after 10k iterations, the generated results are nowhere close to the training data.
Btw this is what the generated results for an empty prompt looks like after 10k iterations.
![image](https://github.com/bghira/SimpleTuner/assets/5581222/c0990941-c7e9-4cc5-85e2-967a0705fedb)

**Comment by bghira** [comment]
i'm guessing the successful training was before the multi-dataset changes. can you show your dataset config?

**Comment by bghira** [comment]
also can you look in the metadata json that is generated for  your dataset, and see whether the crop values etc look 'sane'?

**Comment by janzd** [comment]
This is the dataset config file

```JSON
[
    {
        "id": "DATASET_NAME",
        "type": "local",
        "instance_data_dir": "/data/src/SimpleTuner/data/dataset",
        "crop": false,
        "crop_style": "center",
        "crop_aspect": "preserve",
        "resolution": 1.0,
        "resolution_type": "pixel",
        "minimum_image_size": 1.0,
        "prepend_instance_prompt": false,
        "instance_prompt": "",
        "only_instance_prompt": false,
        "caption_strategy": "filename",
        "cache_dir_vae": "/data/src/SimpleTuner/data/dataset/vaecache",
        "vae_cache_clear_each_epoch": false,
        "probability": 1.0,
        "text_embeds": "dataset_captions"
    },
    {
        "id": "dataset_captions",
        "dataset_type": "text_embeds",
        "default": true,
        "type": "local",
        "cache_dir": "/data/src/SimpleTuner/data/dataset/textembed_cache"
    }
]
```

And this is metadata excerpt
```
{"original_size": [1024, 1024], "crop_coordinates": [0, 0], "target_size": [1024, 1024], "aspect_ratio": 1.0, "luminance": 238.63047203350067}
```
All images are in 1024x1024 resolution, so all instances in the metadata have these same size and crop values.

**Comment by bghira** [comment]
i think the old version wasn't working properly with filename-based captions, maybe? that was something that was fixed in a newer version.

**Comment by janzd** [comment]
I've tried using both actually.

I was wondering, is there any dataset on huggingface for example that I could try that you've tested it with and the model seemed to learn properly?

**Comment by bghira** [comment]
https://huggingface.co/datasets/ptx0/photo-concept-bucket/viewer/default/train

this one contains 568k photos and CogVLM captions (the cogvlm_caption column) and it has been in use for a while

https://huggingface.co/datasets/ptx0/hands-2.7k

this one is useful for making test models because around 1600 steps at a substantial enough learning rate, everything will be made of human hands :D

**Comment by janzd** [comment]
Thanks! I'll try running it with these datasets to see if it works without an issue and if it does then I guess there's something about my dataset or the training configuration that makes it incompatible or hard to use the data to train with the current version of the code.

**Comment by janzd** [comment]
I was busy with other projects but I finally found time to test it.
I tried the hands dataset with 8e-5 learning rate and after a few hundred iterations, I got these results.
This is for an empty prompt.
![hands_empty_prompt_exp3](https://github.com/bghira/SimpleTuner/assets/5581222/65670e67-70bc-4514-b986-0e901df5dd33)
And this is a prompt specifying an image of a closeup of a human hand.
![hands_exp3](https://github.com/bghira/SimpleTuner/assets/5581222/a6402454-6666-49eb-9c4e-26d6ef173205)


I guess it works and trains properly but it has trouble with my dataset now for some reason. I just wonder why it worked before but it doesn't know.

**Comment by bghira** [comment]
you can try full u-net tuning with 'bitfit' freeze strategy. this freezes the weights and allows you to train only the bias.

**Comment by bghira** [comment]
i've pushed bitfit code to master.

in `sdxl-env.sh`, you can set `USE_BITFIT=true`.

**Comment by bghira** [comment]
you should also try

```bash
# Bytedance recommends these be set to "trailing" so that inference and training behave in a more congruent manner.
# To follow the original SDXL training strategy, use "leading" instead, though results are generally worse.
export TRAINING_SCHEDULER_TIMESTEP_SPACING="trailing"
export INFERENCE_SCHEDULER_TIMESTEP_SPACING="trailing"
```

set these to "leading" instead of "trailing".

**Comment by janzd** [comment]
I tried both setting it to "leading" and using bitfit to train only the bias, but the results didn't change at all, unfortunately.

Is there any simple way to check that the vae encodings are meaningful and not corrupted in some way? I did try recreating the vae embedding cache though.

**Comment by bghira** [comment]
i mean, it would be pretty obvious if the vae embeds were bad - i'm not sure why it has changed so much for you.

**Comment by bghira** [comment]
the only thing that stands out is a commit from October (not January) that enabled the LR_WARMUP_STEPS parameter in SDXL. yours is set to 3000 steps, maybe you'll have to change that to 0.

**Comment by Night1099** [comment]
Fix for me was setting "aspect_bucket_rounding" to 1

 2 was causing noise problem seen above

although my whole dataset was 1024x1024 minus a few i didnt catch maybe this was problem

<img width="477" height="139" alt="Image" src="https://github.com/user-attachments/assets/ebfda8bd-ff32-4866-bae2-530c076bf9d8" />

**Comment by bghira** [comment]
oh god, yeah. those ratios will break a lot of models. #1303 added min/max aspect options.
